### PR TITLE
Support userborn

### PR DESCRIPTION
**Pull request by r-vdp**
TODO:
  - [x] undo lock file and inputs change once this landed in unstable
  - [x] merge https://github.com/Mic92/sops-nix/pull/611 first

### CODE REVIEW

**Comment by nyabinary** [comment]
It landed on unstable (userborn) :3

File: pkgs/sops-install-secrets/nixos-test.nix [hunk 1]
```diff
@@ -14,10 +14,17 @@ let
...
+      users.users.example-user = lib.mkMerge [
+        (lib.mkIf (! config.systemd.sysusers.enable) {
+          isNormalUser = true;
+          hashedPasswordFile = config.sops.secrets.test_key.path;
+        })
+        (lib.mkIf config.systemd.sysusers.enable {
+          isSystemUser = true;
+          group = "users";
+          hashedPasswordFile = config.sops.secrets.test_key.path;
+        })
```
Comment (r-vdp) [inline pkgs/sops-install-secrets/nixos-test.nix:26]: sysusers can no longer be used to create normal users, there was an assertion added. Only system users are supported now.

Comment (Mic92) [reply pkgs/sops-install-secrets/nixos-test.nix:26]: Ok.

**Comment by nakoo** [comment]
I've been used this PR for a while now and didn't encounter any issue so far. Is there a chance to be merged?

**Comment by Mic92** [comment]
Also works on my laptop.

**Pull request merged**